### PR TITLE
Fixed bug with WCSAxes grid-line visibility

### PR DESCRIPTION
**Pull request by ayshih**
WCSAxes uses round-trip consistency to determine whether grid-line points should be visible.  However, the tolerance for this check is hard-coded to be 0.1 ~degrees~ radians(!), which is far too tolerant when the natural scale of the coordinate frame is something other than degrees (e.g., arcmin).  This PR makes the tolerance a relative tolerance instead.

To showcase the bug, here's an example using SunPy:
download (4)

And after this PR:
download (5)

### CODE REVIEW

**Comment by pllim** [comment]
Oooo nice graphics! Would need a change log and test. Thanks!

Subpackage maintainer, pls re-milestone as needed.

**Comment by Cadair** [comment]
Seems sane to me.

**Comment by ayshih** [comment]
Changelog entry and test added

**Review by Cadair — APPROVED** [review]

**Comment by hamogu** [comment]
@Cadair You think it looks good and it passed CI. I think it looks good, too, and it's not a super-complicated test. Shall we merge or are we waiting for someone else to review?

**Comment by Cadair** [comment]
Let's merge it, it's a small change.

**Pull request merged**
